server/jetstream_cluster.go

```diff
@@ -1070,16 +1084,16 @@ func (js *jetStream) monitorCluster() {
 			}
 			aq.recycle(&ces)
 		case isLeader = <-lch:
-			// We want to make sure we are updated on statsz so ping the extended cluster.
+			js.processLeaderChange(isLeader)
+
 			if isLeader {
 				s.sendInternalMsgLocked(serverStatsPingReqSubj, _EMPTY_, nil, nil)
-			}
-			js.processLeaderChange(isLeader)
-			if isLeader && !beenLeader {
-				beenLeader = true
-				if n.NeedSnapshot() {
-					if err := n.InstallSnapshot(js.metaSnapshot()); err != nil {
-						s.Warnf("Error snapshotting JetStream cluster state: %v", err)
+				// Install a snapshot as we become leader. We will also send to the cluster.
+				if snap := doSnapshot(); snap != nil {
+					// If we are caught up distibute our current state to followers.
```

Comment:
```suggestion
					// If we are caught up distribute our current state to followers.
```

Reply:
Fixed. Not sure why mispell does not catch those..